File: plain/src/test/java/com/github/daneko/android/plain/WeatherServiceTest.java
package com.github.daneko.android.plain;

import com.squareup.okhttp.mockwebserver.MockResponse;
import com.squareup.okhttp.mockwebserver.MockWebServer;

import org.junit.Rule;
import org.junit.Test;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

import static org.assertj.core.api.Assertions.assertThat;


/**
 *
 */
public class WeatherServiceTest {

    @Rule
    public MockWebServer mockWebServer = new MockWebServer();

    @Test
    public void とりあえず正常系だけ() throws IOException {
        mockWebServer.enqueue(new MockResponse()
                .setResponseCode(200)
                .setBody(responseSample()));

        final String s = mockWebServer.url("/").toString();
        final String expected = WeatherService.currentTokyoWeather(s)
                .toBlocking().first().getMain();
        assertThat(expected).isEqualTo("Drizzle");
    }


    private String responseSample() throws IOException {
        StringBuffer sb = new StringBuffer();
        BufferedReader br = new BufferedReader(
                new InputStreamReader(getClass().getClassLoader().getResourceAsStream("response_sample.json"), "UTF-8"));
        for (int c = br.read(); c != -1; c = br.read()) {
            sb.append((char) c);
        }
        return sb.toString();
    }

}

File: kotlin/src/test/kotlin/com/github/daneko/android/kotlin/WeatherServiceTest.kt
package com.github.daneko.android.kotlin

import com.squareup.okhttp.mockwebserver.*
import org.assertj.core.api.Assertions.*
import org.junit.*


/**

 */
class WeatherServiceTest {

    val _mockWebServer = MockWebServer()

    @Rule public fun mockWebServer() = _mockWebServer

    @Test
    fun とりあえず正常系だけ() {
        mockWebServer().enqueue(MockResponse().setResponseCode(200).setBody(responseSample()))
        val s = mockWebServer().url("/").toString()
        val expected = WeatherService.currentTokyoWeather(s).toBlocking().first().main
        assertThat(expected).isEqualTo("Drizzle")
    }

    private fun responseSample(): String {
        return """
{
  "coord": {
    "lon": -0.13,
    "lat": 51.51
  },
  "weather": [
    {
      "id": 310,
      "main": "Drizzle",
      "description": "light intensity drizzle rain",
      "icon": "09n"
    }
  ],
  "base": "stations",
  "main": {
    "temp": 286.26,
    "pressure": 1007,
    "humidity": 87,
    "temp_min": 285.15,
    "temp_max": 287.15
  },
  "visibility": 10000,
  "wind": {
    "speed": 9.3,
    "deg": 210,
    "gust": 14.4
  },
  "clouds": {
    "all": 75
  },
  "dt": 1447640400,
  "sys": {
    "type": 1,
    "id": 5078,
    "message": 0.0348,
    "country": "GB",
    "sunrise": 1447658385,
    "sunset": 1447690208
  },
  "id": 2643743,
  "name": "London",
  "cod": 200
}

        """
    }

}


